buildrunner/validation/config.py

```diff
@@ -9,68 +9,45 @@
 from typing import Dict, List, Optional, Set, Union
 
 # pylint: disable=no-name-in-module
-from pydantic import BaseModel, validator, ValidationError
+from pydantic import BaseModel, Field, validator, ValidationError
 
-
-class Errors:
-    """ Error class for storing validation errors """
-    class Error:
-        """ Error class for storing validation error """
-        def __init__(self, field: str, message: str):
-            self.field: str = field
-            self.message: Union[str, None] = message
-
-    def __init__(self):
-        self.errors = []
-
-    def add(self, field: str, message: str):
-        """ Add an error """
-        self.errors.append(self.Error(field, message))
-
-    def count(self):
-        """ Return the number of errors """
-        return len(self.errors)
-
-    def __str__(self):
-        return '\n'.join([f'  {error.field}:  {error.message}' for error in self.errors])
-
-    def __repr__(self):
-        return self.__str__()
-
-
-class StepBuild(BaseModel):
-    """ Build model within a step """
-    path: Optional[str]
-    dockerfile: Optional[str]
-    pull: Optional[bool]
-    platform: Optional[str]
-    platforms: Optional[List[str]]
-
-
-class StepPushDict(BaseModel):
-    """ Push model within a step """
-    repository: str
-    tags: Optional[List[str]]
-
-
-class Step(BaseModel):
-    """ Step model """
-    build: Optional[Union[StepBuild, str]]
-    push: Optional[Union[StepPushDict, List[Union[str, StepPushDict]], str]]
-
-    def is_multi_platform(self):
-        """
-        Check if the step is a multi-platform build step
-        """
-        return isinstance(self.build, StepBuild) and \
-            self.build.platforms is not None
+from buildrunner.validation.errors import Errors, add_validation_errors
+from buildrunner.validation.step import Step, StepPushCommitDict
 
 
 class Config(BaseModel):
     """ Top level config model """
+
+    # Unclear if this is actively used
+    class GithubModel(BaseModel, extra='forbid'):
+        """ Github model """
+        endpoint: str
+        version: str
+        username: str
+        app_token: str
+
+    class SSHKey(BaseModel, extra='forbid'):
+        """ SSH key model """
+        file: Optional[str]
```

Comment:
Are these working without assigning "None" to the value? That's surprising to me, I thought newer pydantic versions required it

Reply:
Yes it appears to be working. But I don't quite understand what you are saying. Can you give me an example of what you mean?

Reply:
I've had to do `file: Optional[str] = None` to get it working in some cases (newer pydantic versions I thought)

Reply:
This is using an older version of `pydantic` version `1.10.11` because that is the highest that `python-on-whales` allows currently. That may be why I am able to get away with using that syntax.

Reply:
ah, yes, I knew that